### PR TITLE
CAMEL-11259 - Add Distinct Query support to camel-mongodb

### PR DESCRIPTION
**Pull request by elihusmails**
This pull request is sync'd as of 2055 EST.  This pull request contains new functionality to allow for distinct mongodb queries and also unit tests to support optional query parameters.

### CODE REVIEW

File: components/camel-mongodb/src/main/java/org/apache/camel/component/mongodb/MongoDbProducer.java [hunk 1]
```diff
@@ -293,6 +295,36 @@ private void processAndTransferResult(Object result, Exchange exchange, MongoDbO
...
 
+    private Function<Exchange, Object> createDoDistinct() {
+        return exchange -> {
+            Iterable<String> result = new ArrayList<>();
+            MongoCollection<BasicDBObject> dbCol = calculateCollection(exchange);
+            
+            // get the parameters out of the Exchange Header
+            String distinctFieldName = exchange.getIn().getHeader(MongoDbConstants.DISTINCT_QUERY_FIELD, String.class);
+            BasicDBObject query = null;
+            // do not run around looking for a type converter unless there is a need for it
```
Comment (davsclaus) [inline components/camel-mongodb/src/main/java/org/apache/camel/component/mongodb/MongoDbProducer.java]: This is not needed as Camel will check for null body

File: components/camel-mongodb/src/test/java/org/apache/camel/component/mongodb/MongoDbFindOperationTest.java [hunk 2]
```diff
@@ -158,6 +158,7 @@ public void testFindAllIterationOperation() throws Exception {
             headers.put(MongoDbConstants.NUM_TO_SKIP, numToSkip);
             headers.put(MongoDbConstants.LIMIT, 100);
             Object result = template.requestBodyAndHeaders("direct:findAll", (Object) null, headers);
+            System.out.println(result.getClass());
```
Comment (davsclaus) [inline components/camel-mongodb/src/test/java/org/apache/camel/component/mongodb/MongoDbFindOperationTest.java]: Remove this or use LOG.info

**Comment by elihusmails** [comment]
Updated the code based on your suggestions and also sync'd with the master branch from Came's github repo.

**Comment by davsclaus** [comment]
There is also a camel-mongodb3 component. I wonder if this should also have such a functionality as this? If so you are welcome to work on a new PR for that

**Comment by davsclaus** [comment]
Thanks for the PR it has been merged. Do you mind closing this?

**Pull request closed without merging**